staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4495,8 +4495,12 @@ type LocalObjectReference struct {
 // TypedLocalObjectReference contains enough information to let you locate the
 // typed referenced object inside the same namespace.
 type TypedLocalObjectReference struct {
-	// APIGroup is the group for the resource being referenced
-	APIGroup string `json:"apiGroup" protobuf:"bytes,1,opt,name=apiGroup"`
+	// APIGroup is the group for the resource being referenced.
+	// If APIGroup is not specified, the reference Kind must be PersistentVolumeClaim.
```

Comment:
This is a weird cross-field restriction. Why not require the fields to be specified

Reply:
If the kind is PersistentVolumeClaim, apiGroup will be "".  It's odd to require user to specify like this:  

source:
    name: pvc-test
    kind: PersistentVolumeClaim
    apiGroup: ""